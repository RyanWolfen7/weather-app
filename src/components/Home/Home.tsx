import React from 'react';
import styled from 'styled-components';
import Search from '../Search/Search';

const Home: React.FC = () => {
  return (
    <Container>
      <Title>Enter a City and State</Title>
      <FreeArea height={20} />
      <Search dir={'column'} />
    </Container>
  );
}

export default Home;

const Container = styled.div`
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
  height: 100vh;
  flex-direction: column;
`;

const FreeArea = styled.div`
  height: ${(props: { height: number }) => props.height}px;
`;

const Title = styled.span`
  font-size: 45px;
  color: #fff;
  font-weight: 100;
`;
