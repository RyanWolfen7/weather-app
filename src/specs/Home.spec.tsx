import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Home from '../components/Home/Home';

let home: ShallowWrapper;

beforeAll(() => {
  home = shallow(<Home />);
});

test('Home renders and has the title Enter a City and State', () => {
  expect(home.children().first().text()).toEqual('Enter a City and State');
});

test('Home to match snapshot', () => {
  expect(home).toMatchSnapshot();
});
