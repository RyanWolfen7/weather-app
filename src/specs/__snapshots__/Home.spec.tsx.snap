// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home to match snapshot 1`] = `
<styled.div>
  <styled.span>
    Enter a City and State
  </styled.span>
  <styled.div
    height={20}
  />
  <withRouter(Search)
    dir="column"
  />
</styled.div>
`;
